tensorflow_addons/metrics/cohens_kappa.py

```diff
@@ -0,0 +1,161 @@
+    self.conf_mtx = self.add_weight('conf_mtx',
+                                    shape=(self.num_classes, self.num_classes),
+                                    initializer=tf.initializers.zeros,
+                                    dtype=tf.int32)
+
+  def update_state(self, y_true, y_pred, sample_weight=None):
+    """Accumulates the confusion matrix condition statistics.
+    
+    Args:
+      y1 : array, shape = [n_samples]
+           Labels assigned by the first annotator.
+      y2 : array, shape = [n_samples]
+           Labels assigned by the second annotator. The kappa statistic is
+           symmetric, so swapping ``y1`` and ``y2`` doesn't change the value.
+      sample_weight(optional) : for weighting labels in confusion matrix
+           Default is None. Check tf.math.consfusion_matrix for details 
+    
+    Returns:
+      Update op.
+      
+    """
+    y_true = tf.cast(y_true, dtype=tf.int32)
+    y_pred = tf.cast(y_pred, dtype=tf.int32)
+    
+    if y_true.shape != y_pred.shape:
+      raise ValueError("Number of samples in y_true and y_pred are different")
+    
+    # compute the new values of the confusion matrix
+    new_conf_mtx = confusion_matrix(labels=y_true, 
+                                     predictions=y_pred,
+                                     num_classes=self.num_classes,
+                                     weights=sample_weight)
+  
+    # update the values in the orifinal confusion matrix
+    return self.conf_mtx.assign_add(new_conf_mtx)
+
+  def result(self):    
+    nb_ratings = tf.shape(self.conf_mtx)[0]
+    weight_mtx = tf.ones([nb_ratings, nb_ratings], dtype=tf.int32)
+    
+    # 2. Create a weight matrix
+    if self.weightage is None:
+      diagonal = tf.zeros([5], dtype=tf.int32)
+      weight_mtx = tf.linalg.set_diag(weight_mtx, diagonal=diagonal)
+      weight_mtx = tf.cast(weight_mtx, dtype=tf.float32)
+    
+    else:
+      weight_mtx += tf.range(nb_ratings, dtype=tf.int32)
+      weight_mtx = tf.cast(weight_mtx, dtype=tf.float32)
+
+      if self.weightage=='linear':
+        weight_mtx = tf.abs(weight_mtx - K.transpose(weight_mtx))
+      else:
+        weight_mtx = K.pow((weight_mtx - K.transpose(weight_mtx)), 2)
+      weight_mtx = tf.cast(weight_mtx, dtype=tf.float32)
+    
+    # 3. Get counts
+    actual_ratings_hist = K.sum(self.conf_mtx, axis=1)
+    pred_ratings_hist = K.sum(self.conf_mtx, axis=0)
+    
+    # 4. Get the outer product
+    out_prod = pred_ratings_hist[..., None] * actual_ratings_hist[None, ...]
+    
+    # 5. Normalize the confusion matrix and outer product
+    conf_mtx = self.conf_mtx / K.sum(self.conf_mtx)
+    out_prod = out_prod / K.sum(out_prod)
+    
+    conf_mtx = tf.cast(conf_mtx, dtype=tf.float32)
+    out_prod = tf.cast(out_prod, dtype=tf.float32)
+    
+    # 6. Calculate Kappa score
+    numerator = K.sum(conf_mtx * weight_mtx)
+    denominator = K.sum(out_prod * weight_mtx) 
+    kp = 1-(numerator/denominator)
+    
+    return kp
```

Comment:
We should override `get_config` to serialize related configs.
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/metrics.py#L1205 

Also wonder if we should override `reset_states`. Would you mind testing whether the variables are reset correctly?
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/metrics.py#L200

Reply:
I would add `reset_states` but I didn't get `get_config`thing. Can you elaborate a bit on this?

Reply:
Like this:
```python
class CohensKappa(Metric):
  ....
  def get_config(self):
    config = {
        "num_classes": self.num_classes,
        "weightage": self.weightage,
    }
    base_config = super(CohensKappa, self).get_config()
    return dict(list(base_config.items()) + list(config.items()))
```

Reply:
Added both `get_confi` as well as `reset_states`